src/app/modules/training-event/services/event.service.ts

```diff
@@ -1,21 +1,56 @@
-import { Injectable } from '@angular/core';
+import { Injectable, Inject} from '@angular/core';
+import { timer, of as observableOf } from 'rxjs';
+import { catchError, takeWhile } from 'rxjs/operators';
 import { HttpClient } from '@angular/common/http';
 
 import { environment } from 'src/environments/environment';
 import {
+  CampusEventResponse,
   OffCampusEventRequest,
   OffCampusEventResponse,
 } from 'src/app/interfaces/event';
+import { AUTH_SERVICE, AuthService } from 'src/app/interfaces/auth-service';
+import { PaginatedResponse } from 'src/app/interfaces/paginated-response';
 
 /** Provide services for Event. */
 @Injectable({
   providedIn: 'root'
 })
 export class EventService {
+ /** How often should we query latest notifications. */
+ private REFRESH_INTERVAL = 30 * 1000;
 
-  constructor(
-    private readonly http: HttpClient,
-  ) { }
+
+
+ constructor(
+   private readonly http: HttpClient,
+   @Inject(AUTH_SERVICE) private readonly authService: AuthService,
+ ) {
+   this.authService.authenticationSucceed.subscribe(() => {
+     timer(0, this.REFRESH_INTERVAL).pipe(
+       takeWhile(() => this.authService.isAuthenticated),
+       catchError(() => {
+         return observableOf(null);
+       }),
+     ).subscribe();
+   });
+ }
+
+ getEvent(id: number) {
```

Comment:
Please comment this function, and I think it's better to name it as `getCampusEvent` instead of `getEvent`.